pypdf/generic/_data_structures.py

```diff
@@ -1262,21 +1261,21 @@ def set_data(self, data: bytes) -> None:
     @property
     def operations(self) -> List[Tuple[Any, Any]]:
```

Comment:
Should this now be
```suggestion
    def operations(self) -> List[Tuple[Any, bytes]]:
```
?

I'm asking because this PR broke my project (which manipulates this list directly) even though it passed type checking.

Reply:
Looking at `_parse_content_stream` and `isolate_graphics_state` shows that this might include tuples as well.

Reply:
Hm, I don't see where we'd set the second element of the tuple to anything besides `bytes`, but I'm probably missing something. Can you paste a link?

Anyways, assuming you're right, I'm surprised that doesn't crash here: https://github.com/py-pdf/pypdf/blob/6eeac53523ceb5308f502b734801b5826aa79e7b/pypdf/generic/_data_structures.py#L1251

Reply:
Sorry, I have been mistaken here - I misread your comment and thought you wanted it to be `List[bytes]`. Looking at https://github.com/py-pdf/pypdf/blob/545a79cbcfddf95ebe0c2976e5b673c53e555967/pypdf/generic/_data_structures.py#L1403-L1413 it probably should be `bytes` here indeed.

Apart from this: Could you please elaborate what the actual regression has been? The corresponding setter should have reported a type issue if you used the public API as far as I can see from the snippet shown in this comment above.

Reply:
> Apart from this: Could you please elaborate what the actual regression has been? The corresponding setter should have reported a type issue if you used the public API as far as I can see from the snippet shown in this comment above.

My code is doing something like this:
```python
obj = ContentStream(ArrayObject(), None)
obj.operations.append(([], "q"))
obj.operations.append(([], "Q"))
obj.write_to_stream(stream, None)
```

Reply:
Do you want to submit a corresponding PR which fixes the type hint?

Reply:
Sure! https://github.com/py-pdf/pypdf/pull/2941